march_hardware/src/imotioncube/imotioncube.cpp

```diff
@@ -467,7 +466,7 @@ void IMotionCube::downloadSetupToDrive(SdoSlaveInterface& sdo)
         pos = sw_string_.find(delimiter, old_pos);
         next_token = sw_string_.substr(old_pos, pos - old_pos);
 
-        if (pos - old_pos != delimiter.length())
+        if (pos - old_pos != 0)
```

Comment:
```suggestion
        if (pos != old_pos)
```
And also do the same on line 481
```diff
- old_pos = pos + 1;
+ old_pos = pos + delimiter.length();
```